controllers/storagecluster/exporter.go

```diff
@@ -709,3 +715,86 @@ func createMetricsExporterRolebindings(ctx context.Context, r *StorageClusterRec
 
 	return nil
 }
+
+func createRookCephClusterRolebindings(ctx context.Context,
+	r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	// rookCephMonitorMgrRoleBinding is a cluster rolebinding for monitor mgr
+	rookCephMonitorMgrRoleBinding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-ceph-monitor-mgr",
+			Namespace: instance.Namespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "rook-ceph-monitor-mgr",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind: "ServiceAccount",
+				Name: "rook-ceph-mgr",
+			},
+		},
+	}
+
+	// rookCephMonitorRoleBinding is a cluster rolebinding for rook-ceph monitor
+	rookCephMonitorRoleBinding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-ceph-monitor",
+			Namespace: instance.Namespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "rook-ceph-monitor",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind: "ServiceAccount",
+				Name: "rook-ceph-system",
+			},
+		},
+	}
+
+	var roleBindings = []*rbacv1.ClusterRoleBinding{
+		rookCephMonitorMgrRoleBinding, rookCephMonitorRoleBinding,
+	}
+
+	for _, expectedRoleBinding := range roleBindings {
+		currentClusterRoleBinding := new(rbacv1.ClusterRoleBinding)
+		currentClusterRoleBinding.Name = expectedRoleBinding.Name
+		currentClusterRoleBinding.Namespace = instance.Namespace
+
+		_, err := controllerutil.CreateOrUpdate(ctx, r.Client, currentClusterRoleBinding, func() error {
+			// add expected role reference
+			if currentClusterRoleBinding.CreationTimestamp.IsZero() {
+				// RoleRef is immutable. So inject it only while creating new object.
+				currentClusterRoleBinding.RoleRef = expectedRoleBinding.RoleRef
+			}
+
+			// add expected subjects
+			expectedSubjectsLen := len(expectedRoleBinding.Subjects)
+			currentClusterRoleBinding.Subjects = make([]rbacv1.Subject, expectedSubjectsLen)
+			copy(currentClusterRoleBinding.Subjects, expectedRoleBinding.Subjects)
```

Comment:
What is the difference in this approach vs simply assigning value like below?
```
currentClusterRoleBinding.Subjects = expectedRoleBinding.Subjects
```

Reply:
In go everything is copy based assignment except for `slices`, `pointers` and `map`. They are reference based assignment.
In the above case it is a reference based assignment and (usually) in a reference based assignment the referenced object may change (at a later point of time), thus making the previous assignments invalid or error pron (with unexpected results). But here (in this case) we are not changing the referenced (Subject) array contents, so we can do the above.

Reply:
Thanks for the clear explanation.